parser/parser.go

```diff
@@ -172,6 +228,41 @@ func (p *Parser) parseInclude(directive *gonginx.Directive) *gonginx.Include {
 		panic("include can not have a block, or missing semicolon at the end of include statement")
 	}
 
+	if p.opts.parseInclude {
+		includePath := include.IncludePath
+		if !filepath.IsAbs(includePath) {
+			includePath = filepath.Join(p.configRoot, include.IncludePath)
+		}
+		includePaths, err := filepath.Glob(includePath)
+		if err != nil && !p.opts.skipIncludeParsingErr {
+			panic(err)
+		}
+		for _, includePath := range includePaths {
+			if conf, ok := p.parsedIncludes[includePath]; ok {
+				// same file includes itself? don't blow up the parser
+				if conf == nil {
```

Comment:
@riadafridishibly not sure I understand that part.
I understand that you wanted to prevent recursive inclusion, but I think this also breaks multiple inclusions, like, you have 1 partial conf that you include 2 times in different blocks.
So, I think instead of just checking if it is included we should check if the parent is itself or not, or even better, we can still include with depth limit so we panic when the dept is like `1k` we assume it is recursion or just hit our limit etc. Imagine a `nginx/mime.types` file is included in same file but different blocks. 

We can still live with this, but I think we will need to change this at some point or we just remove this part for now and we handle recursive inclusions in another PR.